server/src/main.rs

```diff
@@ -0,0 +1,45 @@
+use clap::Parser;
+
+mod daemon;
+
+const BINARY_NAME: &str = env!("CARGO_BIN_NAME");
+const LOGIN_KEYRING: &str = "login";
+const LOGIN_KEYRING_PATH: &str = "Downloads/default.keyring";
+// const LOGIN_KEYRING_PATH: &str = ".local/share/keyrings/login.keyring";
+#[cfg(debug_assertions)]
+const SERVICE_NAME: &str = "org.freedesktop.secrets_Devel";
+#[cfg(debug_assertions)]
+const SECRET_SESSION_PREFIX: &str = "/org/freedesktop/secrets_Devel/session/";
+#[cfg(not(debug_assertions))]
+const SECRET_SESSION_PREFIX: &str = "/org/freedesktop/secrets/session/";
+#[cfg(debug_assertions)]
+const SECRET_COLLECTION_PREFIX: &str = "/org/freedesktop/secrets_Devel/collection/";
+#[cfg(not(debug_assertions))]
+const SECRET_COLLECTION_PREFIX: &str = "/org/freedesktop/secrets/collection/";
+
+#[derive(Parser, Debug)]
+#[command(version, about, long_about = None)]
+struct Args {
+    #[arg(short = 'l', long, default_value_t = false)]
+    login: bool,
+}
+
+#[tokio::main]
+async fn main() -> daemon::Result<()> {
+    let args = Args::parse();
+    let mut password = String::new();
+    tracing_subscriber::fmt::init();
+
+    if args.login {
+        password = rpassword::prompt_password("Enter the login password: ").unwrap();
+    }
```

Comment:
So otherwise you are passing an empty string, make the Service::new take a Option<String> maybe? Also this sounds very much like it shouldn't be a string. At worst, it should be a Zerioized<String>

Reply:
Done, thanks!